### PR TITLE
APR-1372: add support for filter fields tracer tagging

### PR DESCRIPTION
**Pull request by lhkowalski**
## Change Notes

- Added support for property `tag_filter_fields_on_tracer` so we can decide to tag the filter fields on Map\Repository\Handler

To explain in plain English, the idea is to have a flag to swtich on/off the use of the global tracer to tag every filter field that was sent to `Map\Repository\Handler`:

- if the tag is `true`, we add the machinery to get the global tracer instance inside the `Handler`, generate a string with the field names and set the tag on the tracer
- if the tag is `false`, the code stays the same as before

To allow this, I used annotation processors to insert the needed bits of code on the right places.

## JIRA Issues
- [APR-1372](https://55places.atlassian.net/browse/APR-1372)

### CODE REVIEW

**Review by nico-spada-virtasant — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -93,6 +93,9 @@ The file MUST be named {ACTORNAME}.prefab.definition.yml and saved under `src/`.
     - Can be one of `complete`, `collection`, `minimal`, `handler` or `repository`
     - This field is optional and defaults to `complete`
     - See [below](#supporting-actor-groups) for more information
+- `tag_filter_fields_on_tracer`
+    - This field is optional and default to `false`
+    - Set to `true` if you want to tag on the default global tracer all the filter fields sent to `Map\Repository\Handler`.
```
Comment (Firtzberg) [inline README.md:98]: I guess when you enable this flag the datadog component should be added to the buildable directory paths.
If so, it might be worth just mentioning it here.

File: src/AnnotationProcessor/Actor/Map/Repository/HandlerInterface.php [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<?php
+declare(strict_types=1);
+
+namespace Neighborhoods\Prefab\AnnotationProcessor\Actor\Map\Repository;
+
+interface HandlerInterface {
```
Comment (Firtzberg) [inline src/AnnotationProcessor/Actor/Map/Repository/HandlerInterface.php:6]: Non-blocking
I think you don't need this interface. Just move the constant into the only implementation.
If there is a reason for this, ignore my comment.

**Review by Firtzberg — APPROVED** [review]
Looking good

**Comment by Firtzberg** [comment]
@lhkowalski I don't know how you're testing this, but you might be interested in having a look at this [Fitness](https://github.com/neighborhoods/PrefabFitness/tree/master/UseCase85) use case.

File: README.md (hunk 1 above)
Comment (lhkowalski) [reply README.md:98]: Sure, great suggestion.

**Comment by lhkowalski** [comment]
> @lhkowalski I don't know how you're testing this, but you might be interested in having a look at this [Fitness](https://github.com/neighborhoods/PrefabFitness/tree/master/UseCase85) use case.

I was testing with the development branch in my `composer.json`, generating sample files. I will try to add some usecase test on this Fitness project.

File: README.md (hunk 1 above)
Comment (lhkowalski) [reply README.md:98]: Done.

File: src/BuildConfigurationInterface.php [hunk 3]
```diff
@@ -95,9 +96,17 @@ public function setConstantMap(\Neighborhoods\Prefab\Constant\MapInterface $cons
...
+    public function getTagFilterFieldsOnTracer();
+
+    public function hasTagFilterFieldsOnTracer(): bool;
+=======
     public function setJsonSerializeMapAsArray($jsonSerializeMapAsArray): BuildConfigurationInterface;
 
     public function getJsonSerializeMapAsArray();
 
     public function hasJsonSerializeMapAsArray(): bool;
+>>>>>>> 8.x
```
Comment (nico-spada-virtasant) [inline src/BuildConfigurationInterface.php]: I just found some merging lines. The rest looks good.

**Comment by alexberryman** [comment]
I support this, but I'm not a Prefab owner so I will not be approving/denying this PR.

File: src/BuildConfigurationInterface.php (hunk 3 above)
Comment (lhkowalski) [reply src/BuildConfigurationInterface.php]: Great catch. Thank you.

**Review by harshpatel991 — APPROVED** [review]
